pkg/driver/controller.go

```diff
@@ -339,49 +368,38 @@ func (cs *controller) DeleteVolume(
 	ctx context.Context,
 	req *csi.DeleteVolumeRequest) (*csi.DeleteVolumeResponse, error) {
 
-	klog.Infof("received request to delete volume {%s}", req.VolumeId)
-
-	var (
-		err error
-	)
-
+	var err error
 	if err = cs.validateDeleteVolumeReq(req); err != nil {
 		return nil, err
 	}
-
 	volumeID := strings.ToLower(req.GetVolumeId())
-
-	// verify if the volume has already been deleted
-	vol, err := lvm.GetLVMVolume(volumeID)
-	if vol != nil && vol.DeletionTimestamp != nil {
-		goto deleteResponse
+	if err = cs.deleteVolume(ctx, volumeID); err != nil {
+		return nil, err
 	}
+	return csipayload.NewDeleteVolumeResponseBuilder().Build(), nil
+}
 
+func (cs *controller) deleteVolume(ctx context.Context, volumeID string) error {
+	klog.Infof("received request to delete volume %q", volumeID)
+	vol, err := lvm.GetLVMVolume(volumeID)
 	if err != nil {
 		if k8serror.IsNotFound(err) {
-			goto deleteResponse
+			return nil
 		}
-		return nil, errors.Wrapf(
-			err,
-			"failed to get volume for {%s}",
-			volumeID,
-		)
+		return errors.Wrapf(err,
+			"failed to get volume for {%s}", volumeID)
 	}
-
-	// Delete the corresponding ZV CR
-	err = lvm.DeleteVolume(volumeID)
-	if err != nil {
-		return nil, errors.Wrapf(
-			err,
-			"failed to handle delete volume request for {%s}",
-			volumeID,
-		)
+	if vol.GetDeletionTimestamp() == nil {
+		if err = lvm.DeleteVolume(volumeID); err != nil {
+			return errors.Wrapf(err,
+				"failed to handle delete volume request for {%s}", vol.GetName())
```

Comment:
any sepcific reason to use `vol.GetName()` here instead of  just `volumeID` here in error,similar to above error ?

Reply:
No specific reason. Probably got carried away with refactoring. I'll update it to keep it consistent.